gapic-generator-cloud/templates/cloud/helpers/method_helper.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "active_support/inflector"
+
+module MethodHelper
+  def method_name method
+    ActiveSupport::Inflector.underscore method.name
+  end
+
+  def method_return_type method
+    return "Google::Gax::Operation" if message_lro? method.output
+
+    message_ruby_type method.output
+  end
+
+  def method_lro? method
+    message_lro? method.output
+  end
+
+  def method_desc _method
+    "TODO"
+  end
+
+  def method_arguments method
+    method.input.fields
+  end
+
+  def method_arg_name arg
+    arg.name
+  end
+
+  def method_arg_type arg
+    if arg.message?
+      message_ruby_type arg.message
+    elsif arg.enum?
+      # TODO: handle when arg message is nil and enum is the type
+      "ENUM(#{arg.enum.name})"
```

Comment:
@landrito I am confused because some Field objects are neither a message or an enum. It would be helpful to go through all these methods with you.

Reply:
Ah yeah. The model needs some polishing. We should add some methods in the Field class that check these types: https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto#L137

Reply:
Is this what the Field’s `label` value refers to?

Reply:
Primitive types aren't messages or enums and they should map to the Ruby equivalent. The `label` field is mostly useful for determining if the field is repeated or singular.

Reply:
If the Field has neither a Message or an Enum, then how do I ask it for the primitive type?

Reply:
Ah, the value is `Field#type`. I'll update this.